druid/src/widget/button.rs

```diff
@@ -14,19 +14,98 @@
 
 //! A button widget.
 
-use crate::widget::prelude::*;
 use crate::widget::{Click, ControllerHost, Label, LabelText};
-use crate::{theme, Affine, Data, Insets, LinearGradient, UnitPoint};
+use crate::{theme, Affine, Color, Data, Insets, LinearGradient, UnitPoint};
+use crate::{widget::prelude::*, KeyOrValue};
 
 // the minimum padding added to a button.
 // NOTE: these values are chosen to match the existing look of TextBox; these
 // should be reevaluated at some point.
 const LABEL_INSETS: Insets = Insets::uniform_xy(8., 2.);
 
+/// This describes the values a Button needs to paint itself
+#[derive(Debug)]
+pub struct Style {
+    /// Border width
+    pub border_width: KeyOrValue<f64>,
+    /// Corner radius
+    pub border_radius: KeyOrValue<f64>,
+    /// Border color
+    pub border_color: KeyOrValue<Color>,
+    /// Whether or not to paint the background as a gradient.
+    /// If false, the background defaults to background_color_a
+    pub background_is_gradient: KeyOrValue<bool>,
+    /// First color of gradient, or default color of background
+    pub background_color_a: KeyOrValue<Color>,
+    /// First color of gradient, or alt color of background
+    pub background_color_b: KeyOrValue<Color>,
+}
+
+impl std::default::Default for Style {
+    fn default() -> Self {
+        Self {
+            border_width: theme::BUTTON_BORDER_WIDTH.into(),
+            border_radius: theme::BUTTON_BORDER_RADIUS.into(),
+            border_color: theme::BORDER_DARK.into(),
+            background_is_gradient: true.into(),
+            background_color_a: theme::BUTTON_LIGHT.into(),
+            background_color_b: theme::BUTTON_DARK.into(),
+        }
+    }
+}
+
+/// The Button widget holds a StyleSheet which can resolve to multiple Styles
+pub trait StyleSheet {
+    /// How a button normally looks
+    fn normal(&self) -> Style;
+    /// How a button looks on mouse hover
+    fn hot(&self) -> Style {
+        let normal = self.normal();
+
+        Style {
+            border_color: theme::BORDER_LIGHT.into(),
+            ..normal
+        }
+    }
+    /// How a button looks on mouse down
+    fn active(&self) -> Style {
+        let hot = self.hot();
+        Style {
+            background_color_a: theme::BUTTON_DARK.into(),
+            background_color_b: theme::BUTTON_LIGHT.into(),
+            ..hot
+        }
+    }
+}
+
+struct Default;
+
+impl StyleSheet for Default {
+    fn normal(&self) -> Style {
+        Style::default()
+    }
+}
+
+impl std::default::Default for Box<dyn StyleSheet> {
+    fn default() -> Self {
+        Box::new(Default)
+    }
+}
+
```

Comment:
Okay I'm actually legit confused about all the default stuff here. Could we just not impl `Default`? what would that cost us?